sqlalchemy_databricks/_dialect.py

```diff
@@ -35,6 +57,14 @@ def get_table_names(self, connection, schema=None, **kw):
             query += " IN " + self.identifier_preparer.quote_identifier(schema)
         return [row[1] for row in connection.execute(query)]
 
+    def has_table(self, connection, table_name, schema=None):
+        """override because Databricks raises a different error when no table exists"""
+        try:
+            self._get_table_columns(connection, table_name, schema)
+            return True
+        except exc.DatabaseError:
```

Comment:
This will cause `has_table` to return `False` incorrectly when possible unrelated DatabaseErrors are raised in `_get_table_columns`. I think it might be necessary to override the `_get_table_columns` method instead and alter the regex to match the expected output from Databricks' exception, which looks like a `databricks.sql.exc.ServerOperationError`.

Particularly there is a section here which looks to be re-raising the `OperationalError` due to regex not matching:

```python
        try:
            # This needs the table name to be unescaped (no backticks).
            rows = connection.execute('DESCRIBE {}'.format(full_table)).fetchall()
        except exc.OperationalError as e:
            # Does the table exist?
            regex_fmt = r'TExecuteStatementResp.*SemanticException.*Table not found {}'
            regex = regex_fmt.format(re.escape(full_table))
            if re.search(regex, e.args[0]):
                raise exc.NoSuchTableError(full_table)
            else:
                raise
```